packages/amplify-cli/src/utils/post-install-initialization.ts

```diff
@@ -40,15 +43,19 @@ export const postInstallInitialization = async () => {
 const resolvePackageRoot = (packageName: string) => {
   const resolveDir = path.parse(require.resolve(packageName)).dir;
   const pathParts = resolveDir.split(path.sep);
-  return pathParts.slice(0, pathParts.indexOf(packageName) + 1).join(path.sep);
+  return pathParts.slice(0, pathParts.indexOf(descopePackageName(packageName)) + 1).join(path.sep);
 };
 
+export const descopePackageName = (packageName: string): string => {
```

Comment:
This logic should live in `pathManager.getAmplifyPackageLibDirPath` so it is consistent when assets are written and read. Likewise, when fetching the assets, searchable mock should pass in it's actual package name to `getAmplifyPackageLibDirPath` to get the correct path back.

Separately, I wonder if it would be better for this transform to remove the `@` symbol and change the `/` to a `-`? That guarantees there won't be any naming collisions

Reply:
Moved this to `amplify-cli-core` and updated the `getAmplifyPackageLibDirPath` to handle the descoping. 
Also made changes to the transformation as requested.